examples/field_formats_example/public/examples/3_creating_custom_format_editor.tsx

```diff
@@ -41,12 +41,12 @@ const ExampleCurrencyFormatEditor: FieldFormatEditor<{ currency: string }> = (pr
 ExampleCurrencyFormatEditor.formatId = ExampleCurrencyFormat.id;
 
 // 3. Wrap editor component in a factory. This is needed to support and encourage code-splitting.
-const ExampleCurrencyFormatEditorFactory: FieldFormatEditorFactory<{
-  currency: string;
-}> = async () => ExampleCurrencyFormatEditor;
+const ExampleCurrencyFormatEditorFactory = async () => ExampleCurrencyFormatEditor;
 ExampleCurrencyFormatEditorFactory.formatId = ExampleCurrencyFormatEditor.formatId;
 
 export function registerExampleFormatEditor(indexPatternFieldEditor: IndexPatternFieldEditorSetup) {
   // 4. Register a field editor. This should happen in setup plugin lifecycle phase.
-  indexPatternFieldEditor.fieldFormatEditors.register(ExampleCurrencyFormatEditorFactory);
+  indexPatternFieldEditor.fieldFormatEditors.register(
+    ExampleCurrencyFormatEditorFactory as FieldFormatEditorFactory
```

Comment:
This is my fault and kind of lame but I'm not sure what to do about it.

Reply:
The pattern that comes to mind is to have a new method on the `PluginSetup.fieldFormatEditors` API to control how the consumer creates the factory:
```
export function registerExampleFormatEditor(indexPatternFieldEditor: IndexPatternFieldEditorSetup) {
  // 3. Wrap editor component in a factory. This is needed to support and encourage code-splitting.
  const ExampleCurrencyFormatEditorFactory =
    indexPatternFieldEditor.fieldFormatEditors.createFormatEditorFactory(
      async () => ExampleCurrencyFormatEditor
    );

  // 4. Register a field editor. This should happen in setup plugin lifecycle phase.
  indexPatternFieldEditor.fieldFormatEditors.register(ExampleCurrencyFormatEditorFactory);
}
```

The goal would be to have ExampleCurrencyFormatEditorFactory extend an interface that is accepted by `fieldFormatEditors.register`. I'm not sure right now how much effort it would take to smooth out the underlying parts.

Reply:
>I'm not sure right now how much effort it would take to smooth out the underlying parts.

I mocked some code and I think the strategy being used here is reasonable but there are many layers of code so it can be hard to tell where the problem is. IMO the code is pretty good and other places need more attention than this. YMMV.